tests/unit_tests/instancetype_test.go

```diff
@@ -0,0 +1,178 @@
+	if err != nil {
+		return fmt.Errorf(instanceTypeErrorMessage, labelName, instanceType.Name)
+	}
+
+	if instanceType.Spec.CPU.Guest == uint32(expectedCPU) {
+		return nil
+	}
+
+	return fmt.Errorf(instanceTypeErrorMessage, labelName, instanceType.Name)
+}
+
+func checkMemory(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	memory, err := resource.ParseQuantity(labelValue)
+	if err != nil {
+		return err
+	}
+
+	if instanceType.Spec.Memory.Guest == memory {
+		return nil
+	}
+
+	return fmt.Errorf(instanceTypeErrorMessage, labelName, instanceType.Name)
+}
+
+func checkHugepages(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	boolValue, err := strconv.ParseBool(labelValue)
+	if err != nil {
+		return err
+	}
+
+	if instanceType.Spec.Memory.Hugepages != nil && boolValue {
+		return nil
+	}
+
+	return fmt.Errorf(instanceTypeErrorMessage, labelName, instanceType.Name)
+}
+
+func checkGPUs(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	return checkAttribute(labelValue, labelName, instanceType.Name, instanceTypeErrorMessage, instanceType.Spec.GPUs != nil)
+}
+
+func checkNuma(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	return checkAttribute(labelValue, labelName, instanceType.Name, instanceTypeErrorMessage, instanceType.Spec.CPU.NUMA != nil)
+}
+
+func checkDedicatedCPUPlacement(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	return checkAttribute(labelValue, labelName, instanceType.Name, instanceTypeErrorMessage, instanceType.Spec.CPU.DedicatedCPUPlacement != nil)
+}
+
+func checkIsolateEmulatorThread(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	return checkAttribute(labelValue, labelName, instanceType.Name, instanceTypeErrorMessage, instanceType.Spec.CPU.IsolateEmulatorThread != nil)
+}
+
+func checkDisplayName(labelValue, labelName string, preference instancetypev1beta1.VirtualMachineClusterPreference) error {
+	return checkAttribute(labelValue, labelName, preference.Name, instanceTypeErrorMessage, preference.Name == labelValue)
+}
+
+func preferenceCheckVendor(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterPreference) error {
+	return checkAttribute(labelValue, labelName, instanceType.Name, preferenceErrorMessage, labelValue == expectedVendor)
+}
+
+func instancetypeCheckVendor(labelValue, labelName string, instanceType instancetypev1beta1.VirtualMachineClusterInstancetype) error {
+	return checkAttribute(labelValue, labelName, instanceType.Name, instanceTypeErrorMessage, labelValue == expectedVendor)
+}
+
+func checkAttribute(labelValue, labelName, resourceName, errorMessage string, condition bool) error {
```

Comment:
Inline this func? It doesn't provide much value.

Reply:
I don't see it doing much other than formatting the error. Can you inline it please? Also can you inverse the conditions please? So the inlined code should look like this:

```
if condition {
    return fmt.Errorf(errorMessage, labelName, resourceName)
}

return nil
```